src/client/Microsoft.Identity.Client/ManagedIdentity/ManagedIdentitySource.cs

```diff
@@ -0,0 +1,90 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Linq;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Identity.Client.Extensibility;
+using Microsoft.Identity.Client.Http;
+using Microsoft.Identity.Client.Utils;
+using Microsoft.Identity.Client.Internal;
+using Microsoft.Identity.Client.Core;
+
+namespace Microsoft.Identity.Client.ManagedIdentity
+{
+    internal abstract class ManagedIdentitySource
+    {
+        internal const string AuthenticationResponseInvalidFormatError = "Invalid response, the authentication response was not in the expected format.";
+        internal const string UnexpectedResponse = "Managed Identity response was not in the expected format. See the inner exception for details.";
+        internal RequestContext _requestContext;
+
+        protected ManagedIdentitySource(RequestContext requestContext)
+        {
+            _requestContext = requestContext;
+        }
+
+        protected internal string ClientId { get; }
+
+        public virtual async Task<ManagedIdentityResponse> AuthenticateAsync(AppTokenProviderParameters parameters, CancellationToken cancellationToken)
+        {
+            ManagedIdentityRequest request = CreateRequest(parameters.Scopes.ToArray());
+            
+            var response =
+            HttpMethod.Get.Equals(request.Method) ?
```

Comment:
nit:
```suggestion
            request.Method.Equals(HttpMethod.Get) ?
```

Reply:
Generally in Java it is a practice to keep the constants on left in the comparison. Maybe not in C#.

Reply:
It's an old practice from C to avoid using `=` instead of `==`, since `if (NULL= x)` fails but `if(x = NULL)` works in C. https://stackoverflow.com/questions/370366/why-put-the-constant-before-the-variable-in-a-comparison

Not sure it affects Java. Does not affect C#

<img width="135" alt="image" src="https://user-images.githubusercontent.com/12273384/204104360-7cdcf36a-6cf0-4805-a4d3-73d636844e40.png">